lib/dry/system/component.rb

```diff
@@ -41,88 +37,29 @@ class Component
       #   @return [Hash] component's options
       attr_reader :options
 
-      # Returns a component with a namespace and path provided from a file found within
-      # the given component dirs. If no file is found, a component is returned without
-      # these attributes.
-      #
-      # @return [Dry::System::Component]
-      # @api private
-      def self.locate(identifier, component_dirs, options = EMPTY_HASH)
-        options = DEFAULT_OPTIONS.merge(options)
-
-        path = identifier.to_s.gsub(options[:separator], PATH_SEPARATOR)
-
-        found_dir, found_path = component_dirs.detect { |dir|
-          if (component_file = dir.component_file(path))
-            break [dir, component_file]
-          end
-        }
-
-        return new(identifier, options) unless found_path
-
-        new_from_component_dir(identifier, found_dir, found_path, options)
-      end
-
-      # @api private
-      def self.new_from_component_dir(identifier, component_dir, file_path, options = EMPTY_HASH)
-        # Replace default options (provided via args) with more component-local values
-        options = {
-          **DEFAULT_OPTIONS,
-          **options,
-          **component_dir.component_options,
-          **MagicCommentsParser.(file_path)
-        }
-
-        new(
-          identifier,
-          namespace: component_dir.default_namespace,
-          file_path: file_path,
-          **options
-        )
-      end
-
       # @api private
       def self.new(identifier, options = EMPTY_HASH)
         options = DEFAULT_OPTIONS.merge(options)
 
-        namespace, separator = options.values_at(:namespace, :separator)
-
-        identifier = extract_identifier(identifier, namespace, separator)
-
-        path = identifier.gsub(separator, PATH_SEPARATOR)
-        if namespace
-          namespace = namespace.to_s.gsub(separator, PATH_SEPARATOR)
-          path = "#{namespace}#{PATH_SEPARATOR}#{path}"
-        end
-
-        super(identifier, path: path, **options)
-      end
-
-      # @api private
-      def self.extract_identifier(identifier, namespace, separator)
-        identifier = identifier.to_s
-
-        identifier = namespace ? remove_namespace_from_name(identifier, namespace) : identifier
-
-        identifier.scan(WORD_REGEX).join(separator)
-      end
-      private_class_method :extract_identifier
-
-      # @api private
-      def self.remove_namespace_from_name(name, namespace)
-        match_value = name.match(/^(?<remove_namespace>#{namespace})(?<separator>\W)(?<identifier>.*)/)
+        identifier =
+          if identifier.is_a?(Identifier)
+            identifier
+          else
+            Identifier.new(
+              identifier,
+              namespace: options.delete(:namespace),
+              separator: options.delete(:separator)
```

Comment:
@flash-gordon Thanks for pointing this out. I considered it when writing this, and I thought it would be no real ill consequences of "leaking" those spurious options to the initializer. But I think you're right that this should be handled properly. I'll push up a fix for this before merging.